saleor/payment/gateways/braintree/__init__.py

```diff
@@ -141,28 +141,75 @@ def authorize(
         raise BraintreeException(DEFAULT_ERROR_MESSAGE)
 
     gateway_response = extract_gateway_response(result)
+
     error = get_error_for_client(gateway_response["errors"])
     kind = TransactionKind.CAPTURE if config.auto_capture else TransactionKind.AUTH
     credit_card = gateway_response.get("credit_card", {})
+
+    if(error):
+        print('error occured')
+        return GatewayResponse(
```

Comment:
A lot of code seems to be duplicated when constructing the gateway response objects here. Maybe we could create first a gateway response with fields that are common to all cases, and then inside the `if` conditions only provide fields specific to these particular cases?